rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,234 @@
+from icon_rapid7_insight_agent.util.graphql_api.region_map import region_map
+from icon_rapid7_insight_agent.util.graphql_api.api_exception import APIException
+import icon_rapid7_insight_agent.util.graphql_api.agent_typer as agent_typer
+import requests
+
+class ApiConnection():
+    def __init__(self, api_key, org_key, region_string, logger):
+        self.api_key = api_key
+        self.org_key = org_key
+        self.logger = logger
+
+        self.endpoint = self._setup_endpoint(region_string)
+
+        self.session = requests.Session()
+        self.session.headers = self._get_headers()
+
+    def get_agent(self, agent_input):
+        agent_type = agent_typer.get_agent_type(agent_input)
+        agents = self._get_all_agents()
+        agent = self._find_agent_in_agents(agents, agent_input, agent_type)
+        return agent
+
+    def quarantine(self, advertisement_period, agent_id, quarantine_state):
+        """
+        Take a quarantine action on a given agent id
+
+        :param advertisement_period: int (Amount of time in seconds to try to take the quarantine action)
+        :param agent_id: string
+        :param quarantine_state: boolean (quarantine or unquarantine the agent)
```

Comment:
```suggestion
        :param quarantine_state: boolean (quarantine or unquarantine the agent)
```
True = quarantine, False = unquarantine? That should be documented